plugins/send2ebook.koplugin/main.lua

```diff
@@ -0,0 +1,229 @@
+        logger.dbg("Send2Ebook: obtaining download folder")
+        send2ebook_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), send2ebook_config_file))
+        if send2ebook_settings:has(config_key_custom_dl_dir) then
+            download_dir_path = send2ebook_settings:readSetting(config_key_custom_dl_dir)
+        else
+            download_dir_path = ("%s/%s/"):format(DataStorage:getFullDataDir(), default_download_dir_name)
+        end
+
+        if not lfs.attributes(download_dir_path, "mode") then
+            logger.dbg("Send2Ebook: Creating initial directory")
+            lfs.mkdir(download_dir_path)
+        end
+    end
+end
+
+function Send2Ebook:process()
+    local info = InfoMessage:new{ text = _("Connecting …") }
+    UIManager:show(info)
+    logger.dbg("Send2Ebook: force repaint due to upcoming blocking calls")
+    UIManager:forceRePaint()
+    UIManager:close(info)
+
+    local count = 1
+    local ftp_config = send2ebook_settings:readSetting("ftp_config")
+
+    local connection_url = FtpApi:generateUrl(ftp_config.address, ftp_config.username, ftp_config.password)
+
+    local fileTable = FtpApi:listFolder(connection_url .. ftp_config.folder, ftp_config.folder) --args looks strange but otherwise resonse with invalid paths
+
+    if not fileTable then
+      info = InfoMessage:new{ text = T(_("Could not get file list for server: %1, user: %2, folder: %3"), ftp_config.address, ftp_config.username, ftp_config.folder) }
+    else
+      local total_entries = table.getn(fileTable)
+      logger.dbg("total_entries ", total_entries)
+      if total_entries > 1 then total_entries = total_entries -2 end --remove result "../" (upper folder) and "./" (current folder)
+      for idx, ftpFile in ipairs(fileTable) do
+          logger.dbg("ftpFile ", ftpFile)
+          if ftpFile["type"] == "file" and stringEnds(ftpFile["text"], supported_ftp_file_type) then
+
+              info = InfoMessage:new{ text = T(_("Processing %1/%2"), count, total_entries) }
+              UIManager:show(info)
+              UIManager:forceRePaint()
+              UIManager:close(info)
+
+              local remote_file_path = ftpFile["url"]
+              logger.dbg("remote_file_path", remote_file_path)
+              local local_file_path = download_dir_path .. ftpFile["text"]
+              count = count + Send2Ebook:downloadFileAndRemove(connection_url, remote_file_path, local_file_path)
+              end
+          info = InfoMessage:new{ text = T(_("Processing finished. Success: %1"), count -1) }
+          end
+    end
+    UIManager:show(info)
+    Send2Ebook:afterWifiAction()
+end
+
+function Send2Ebook:removeReadActicles()
+    logger.dbg("Send2Ebook: Removing read articles from :", download_dir_path)
+    for entry in lfs.dir(download_dir_path) do
+        if entry ~= "." and entry ~= ".." then
+            local entry_path = download_dir_path .. entry
+            if DocSettings:hasSidecarFile(entry_path) then
```

Comment:
Does this pr https://github.com/koreader/koreader/pull/3675/files
Make any influence to above check? Cause after merge master it doesn't seem to work correctly, which it did on emulator (and old master)

Reply:
Maybe I should use something else here?

Reply:
It seems to be working fine on emulator, after todays master merge